docs/source/integrations/mlflow.md

```diff
@@ -0,0 +1,278 @@
+consider disabling them to avoid logging the runs twice.
+:::
+
+`kedro-mlflow` will use a subset of the `run_params` as tags for the MLflow run,
+and log the Kedro parameters as MLflow parameters:
+
+```{image} ../meta/images/complete-mlflow-tracking-kedro-mlflow.png
+:alt: Complete MLflow tracking with kedro-mlflow
+:width: 50%
+:align: center
+```
+
+Check out {external+kedro-mlflow:doc}`the official kedro-mlflow tutorial <source/03_getting_started/02_first_steps>`
+for more detailed steps.
+
+Notice that `kedro-mlflow` assumes 1 Kedro project = 1 MLflow experiment.
+If this does not suit your use case, consider pursuing a more customised approach
+like the one described in the previous section.
+
+### Tracking Kedro in MLflow using the Python API
+
+If you are running Kedro programmatically using the Python API,
+you can log your runs using the MLflow "fluent" API.
+
+For example, taking the {doc}`lifecycle management example </kedro_project_setup/session>`
+as a starting point:
+
+```python
+from pathlib import Path
+
+import mlflow
+from kedro.framework.session import KedroSession
+from kedro.framework.startup import bootstrap_project
+
+bootstrap_project(Path.cwd())
+
+mlflow.set_experiment("Kedro Spaceflights test")
+
+with KedroSession.create() as session:
+    with mlflow.start_run():
+        mlflow.set_tag("session_id", session.session_id)
+        session.run()
+```
+
+If you want more flexibility or to log extra parameters,
+you might need to run the Kedro pipelines manually yourself.
+
+### Simple artifact tracking in MLflow using Hooks
+
+MLflow also helps you store "artifacts" associated with a run,
+which include compressed model weights, images, and other typically large files.
+
+For example, you can use the {py:meth}`~kedro.framework.hooks.specs.DatasetSpecs.before_dataset_saved` Hook
+to store a pickled version of the data, as follows:
+
+```python
+import pickle
+import tempfile
+import typing as t
+from pathlib import Path
+
+
+class ArtifactTrackingHooks:
+    @hook_impl
+    def before_dataset_saved(self, dataset_name: str, data: t.Any):
+        logger.info(f"Logging dataset '{dataset_name}' to MLFlow")
+
+        # Serialise object to temporary directory and log it to MLFlow
+        with tempfile.TemporaryDirectory() as temp_dir:
+            file_path = Path(temp_dir) / f"{dataset_name}.pkl"
+            try:
+                with open(file_path, "wb") as fh:
+                    pickle.dump(data, fh)
+            except Exception:
+                logger.error(f"Failed to pickle dataset '{dataset_name}'")
+            else:
+                mlflow.log_artifact(file_path)
+```
+
+Resulting in something like this:
+
+```{image} ../meta/images/mlflow-pickle-artifact.png
+:alt: Tracking artifacts as pickles in MLflow
+:width: 50%
+:align: center
+```
+
+Notice though that storing artifacts as Python pickles
+prevents you from fully using MLflow native preview capabilities.
```

Comment:
Is it really necessary to specify it? Just a nitpick, but maybe it would be nicer to write it the other way around with a csv or a png example which would render with a nice preview and say that mlflow offers a preview for some data format?